### ISSUE TITLE
NotImplementedError: It is not support to use $id with $ref

### ISSUE THREAD

**Issue by edwardpmorris** [koxudaxi/datamodel-code-generator#256]
**Is your feature request related to a problem? Please describe.**

An Error occurs "NotImplementedError: It is not support to use $id with $ref" when attempting to convert local JSON schema file with multiple `$ref`. Did not find any mention of this limitation, or a description of whether remote files are supported in the docs.

For example when converting https://vizztools.github.io/vizzToolsCore/json-schema/Organization.schema.json using 

```
datamodel-codegen  --input json-schema/Organization.schema.json --input-file-type jsonschema --output vizzToolsCore/Organization.py
```  

**Describe the solution you'd like**

Would like to be able to convert to Pydantic types using JSON schema that use $id to ["declare a base URI against which $ref URI-references are resolved"](https://json-schema.org/understanding-json-schema/structuring.html?highlight=definitions#the-id-property)

**Describe alternatives you've considered**

If this is tricky to implement, maybe some examples and tips on how to optimally structure complicated multi-file schema to use with `datamodel-code-generator`. 

**Additional context**

This is a great tool! Very interested in using Pydantic and `datamodel-code-generator` as the main generator to build our Python data models.

**Comment by koxudaxi** [comment]
@edwardpmorris 
Thank you for creating this issue.
I'm working on implementing $id.
I have released a new version as `0.6.5`
This version supports $id and a root $id
But, your JSONSchema can not be parsed. because your schema uses the special character(`@` ) and complex referenced structure. 
I want to improve the parser.

**Comment by edwardpmorris** [comment]
thanks @kouxudai . oops my bad with "@id", will give the new version a try with the error in my schema corrected.

**Comment by edwardpmorris** [comment]
After fixing the property `$id` error. Using local file `$refs`. A quick test with `datamodel-codegen --version 0.6.5` gives `NotADirectoryError: [Errno 20] Not a directory: '/home/edward/Documents/vizzToolsCore/json-schema/Organization.schema.json/@id.schema.json'`

Looks like the basename `Organization.schema.json` of `$id` needs to removed during the parsing?

**Comment by koxudaxi** [comment]
Interesting.
Would you please give me the files?
I expected only remote file with $id

**Comment by edwardpmorris** [comment]
Thinking that using local file refs might help with the code generation; I changed to only update `$id` when deployed to gh-pages. 

Here are the originals: 
[Organization_schema_example.zip](https://github.com/koxudaxi/datamodel-code-generator/files/5529942/Organization_schema_example.zip)
The online version is https://vizztools.github.io/vizzToolsCore/json-schema/Organization.schema.json

Admittedly I may have misinterpreted how `$id` and `$ref` should be used.

**Comment by koxudaxi** [comment]
@edwardpmorris 
Thank you for sharing your schemas.
I have fixed the code to parse for your schemas.
Would you please check the generated code and modified schemas?
https://github.com/koxudaxi/datamodel-code-generator/pull/261/files.

I removed `@` and `$` from the model names and attributes.
This code generator doesn't parse $id and $ref perfectly.
But, The code generator covers use-cases.

**Comment by koxudaxi** [comment]
I released a new version( `0.6.8`) includes the PR.
